source/Server/WorkItems/WorkItemLinkMapper.cs

```diff
@@ -21,14 +21,13 @@ public WorkItemLinkMapper(IAzureDevOpsConfigurationStore store, IAdoApiClient cl
         public string CommentParser => AzureDevOpsConfigurationStore.CommentParser;
         public bool IsEnabled => store.GetIsEnabled();
 
-        public SuccessOrErrorResult<WorkItemLink[]> Map(OctopusPackageMetadata packageMetadata)
+        public SuccessOrErrorResult<WorkItemLink[]> Map(OctopusBuildInformation buildInformation)
         {
             if (!IsEnabled
-                || string.IsNullOrWhiteSpace(packageMetadata?.BuildUrl)
-                || packageMetadata.CommentParser != CommentParser)
+                || string.IsNullOrWhiteSpace(buildInformation?.BuildUrl))
                 return null;
```

Comment:
I think it might be a good idea to check `BuildEnvironment == "Azure DevOps"` here to replace the retired `CommentParser` guard. Otherwise, I expect the issue tracker to send API requests to unexpected build environments.

I'm less concerned about the equivalent Jira and GitHub false positives, because they're at a relatively low rate, and they involve probing a known endpoint in a known-valid way.